lib/bundle.js

```diff
@@ -20,158 +20,253 @@ var Builder = require('systemjs-builder');
-
-    operations.push({
-      operator: operator,
-      moduleName: moduleName
+  var seq = expandGlob(args[0])
+    .then(function(moduleNames) {
+      pushOps(moduleNames, '+');
     });
+
+  for (i = 1; i < args.length - 1; i += 2) {
+    seq = (function(name, operator) {
+      return seq.then(function() {
+        return expandGlob(name)
+          .then(function(moduleNames) {
+            pushOps(moduleNames, operator)
+          });
+      });
+    })(args[i + 1], args[i]);
   }
 
-  return operations;
+  seq = seq.then(function() {
+    return operations;
+  });
+
+  return seq;
+
+  function pushOps(moduleNames, operator) {
+    moduleNames.forEach(function(mn) {
+      operations.push({
+        moduleName: mn,
+        operator: operator
+      });
+    });
+  }
 }
 
 function extractBundleName(fileName) {
   return path.relative(config.pjson.baseURL, fileName.replace(/\.js$/, '')).replace(/\\/g, '/');
 }
 
+function getModuleName(file) {
+  var baseURL = config.pjson.baseURL + '/';
+  var paths = systemBuilder.loader.paths;
+
+  var id = '';
+  var matchLength = 0;
+
+  for (rule in paths) {
+    var resolvedPath = path.resolve(baseURL, paths[rule]).replace(/\\/g, '/');
+    var pathDir = path.dirname(resolvedPath);
+    var fileDir = path.dirname(file);
+    var pathLength = pathDir.split('/').length;
+
+    var regex = new RegExp('^' + pathDir, 'i');
+
+    if (regex.test(fileDir) && matchLength < pathLength) {
+      var parts = rule.split('*');
+      var moduleName = file.replace(regex, '').replace(/^\//, '').replace(/\.js$/, '');
+      id = parts[0] + moduleName;
+      matchLength = pathLength;
+    }
+  }
+
+  if (id === '') {
+    throw new Error('No matching path defined for file: ' + "'" + file + "'");
+  }
+
+  return id;
+}
+
+function expandGlob(moduleName) {
+  var loader = systemBuilder.loader;
+  return loader.normalize(moduleName)
+    .then(function(normalized) {
+      return loader.locate({
+        name: normalized
+      });
+    })
+    .then(function(address) {
+      var normalizedAddress = path.normalize(address).replace(/\\/g, '/').replace(/^file:/, '');
+
+      var files = glob.sync(normalizedAddress, {
+        nobrace: true,
+        noext: true,
+        nonegate: true,
+        nocomment: true,
+        nodir: true
+      });
+
+      var modules = files.map(function(file) {
+        return getModuleName(file);
+      });
+      return modules;
+    });
+}
+
+function getOperatorFunc(op) {
+  var operatorFunction;
+  if (op.operator == '+')
+    operatorFunction = systemBuilder.addTrees;
+  else if (op.operator == '-')
+    operatorFunction = systemBuilder.subtractTrees;
+  else
+    throw 'Unknown operator ' + op.operator;
+  return operatorFunction;
+}
+
 // options.inject, options.sourceMaps, options.minify
 exports.bundle = function(moduleExpression, fileName, opts) {
-  var args = moduleExpression.split(' ');
 
-  var firstModule = args[0];
+  var regex = /\?|\!|\[|\]|{|}|\|/;
```

Comment:
`!` is allowed in module names - for example `jspm bundle file.css!`. Also `{` and `}` will be supported as well for conditionals.

Perhaps rather than throwing outright, just ensure the characters are simply escaped before putting into the glob function?